### PR TITLE
Replace the specification of frequency for hours in pandas data_range

### LINKED ISSUE #1046: Pandas gives a future warning for usage of "H" in a date_range
https://github.com/oemof/oemof-solph/blob/544ffc8ee21f257b475030bc4a1a41e27279d3cd/src/oemof/solph/_energy_system.py#L305

The capital `H` should be replaced by smaller case `h` here, I checked with the current minimum pandas version dependency (1.5.3), it does not seem to interfere with it. I will open an PR.

### PR DESCRIPTION
**Pull request by fwitte**
Closes #1046

### CODE REVIEW

File: .github/workflows/tox_pytests.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8, 3.9, "3.10"]
+        python-version: [3.9, "3.10", "3.11"]
```
Comment (p-snft) [inline .github/workflows/tox_pytests.yml:19]:
```suggestion
        python-version: [3.9, "3.10", "3.11", "3.12"]
```
We should definitely support the current stable Python version, which is 3.12. I don't know, if we should even continue supporting Python 3.9: Debian 12 (current stable) has Python 3.11 already, Ubuntu 22.04 has Python 3.10 as default.

**Review by p-snft — APPROVED** [review]
This is a required hotfix. Maybe, supporting different versions of Python is something for another PR.

**Comment by fwitte** [comment]
What is the current state here? I am getting tired of the pandas warnings... :D

**Comment by p-snft** [comment]
Feel free to merge. (Didn't manage to get GitHub to test using Python 3.12 but that is a separate issue.)

**Comment by fwitte** [comment]
Thank you, merged it!

**Pull request merged**